server/src/parser/common/types/index.ts

```diff
@@ -398,6 +398,11 @@ export interface ISolFileEntry {
    * Has an analysis pass succeeded, allowing a user to operate on the ast.
    */
   isAnalyzed(): boolean;
+
+  /**
+   * Local means the file is part of the project's scope, and not external, dependency or library
+   */
+  isLocal: boolean;
```

Comment:
I am not sure about `isLocal` as the name. It really indicates that a file should be primed and preanalysed during initialization, or is there more that we intend to do based on this property?

Reply:
Pato has confirmed that is how the concept is named in HH. So no need to change.